components/style/properties/properties.mako.rs

```diff
@@ -2588,6 +2591,20 @@ pub fn apply_declarations<'a, F, I>(device: &Device,
     style.build()
 }
 
+/// See StyleAdjuster::adjust_for_auto_minsize.
+% if product == "servo":
+pub fn adjust_auto_minsize(style: &mut StyleBuilder, is_item: bool) {
+    % for direction in ["width", "height"]:
+        // Like calling to_computed_value, which wouldn't type check.
+        if style.get_position().clone_min_${direction}() == computed::LengthOrPercentageOrAuto::Auto
+            && !is_item {
+                style.mutate_position().set_min_${direction}(
+                    computed::LengthOrPercentageOrAuto::Length(Au(0))
```

Comment:
My last try failed to find where the spec asked to return `0px` in normal element ...